test/e2e/syncer/spec_test.go

```diff
@@ -0,0 +1,254 @@
+				nsExpecter := framework.NewExpecter(nsInformer.Informer())
+				sinkNsLister := nsInformer.Lister()
+				coreSharedInformerFactory.Start(ctx.Done())
+
+				wildwestSharedInformerFactory := wildwestexternalversions.NewSharedInformerFactoryWithOptions(servers[sinkClusterName].wildwestClientSet, 0)
+				cowboyInformer := wildwestSharedInformerFactory.Wildwest().V1alpha1().Cowboys()
+				cowboyExpecter := framework.NewExpecter(cowboyInformer.Informer())
+				sinkCowboyLister := cowboyInformer.Lister().Cowboys("")
+				wildwestSharedInformerFactory.Start(ctx.Done())
+
+				ns, err := sourceNsClient.Create(ctx, &corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: testNamespace,
+					},
+				}, metav1.CreateOptions{})
+				if err != nil {
+					t.Errorf("failed to create namespace: %v", err)
+					return
+				}
+
+				cowboy, err := sourceCowboyClient.Create(ctx, &wildwestv1alpha1.Cowboy{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "timothy",
+					},
+					Spec: wildwestv1alpha1.CowboySpec{Intent: "yeehaw"},
+				}, metav1.CreateOptions{})
+				if err != nil {
+					t.Errorf("failed to create cowboy: %v", err)
+					return
+				}
+
+				if !cache.WaitForNamedCacheSync(t.Name(), ctx.Done(), nsInformer.Informer().HasSynced) {
+					t.Error("failed to wait for namespace cache to sync")
+					return
+				}
+
+				if !cache.WaitForNamedCacheSync(t.Name(), ctx.Done(), cowboyInformer.Informer().HasSynced) {
+					t.Error("failed to wait for cowboy cache to sync")
+					return
+				}
+
+				defer servers[sourceClusterName].Artifact(t, func() (runtime.Object, error) {
+					return sourceNsClient.Get(ctx, ns.Name, metav1.GetOptions{})
+				})
+				defer servers[sourceClusterName].Artifact(t, func() (runtime.Object, error) {
+					return sourceCowboyClient.Get(ctx, cowboy.Name, metav1.GetOptions{})
+				})
+				defer servers[sinkClusterName].Artifact(t, func() (runtime.Object, error) {
+					return sinkNsClient.Get(ctx, fmt.Sprintf("kcp--admin--%s", ns.Name), metav1.GetOptions{})
+				})
+				defer servers[sinkClusterName].Artifact(t, func() (runtime.Object, error) {
+					return sinkWildwestClient.Cowboys(fmt.Sprintf("kcp--admin--%s", cowboy.Name)).Get(ctx, cowboy.GetName(), metav1.GetOptions{})
+				})
+
+				if err := nsExpecter.ExpectBefore(ctx, func(ctx context.Context) (bool, error) {
+					sinkNsList, err := sinkNsLister.List(labels.Everything())
+					if err != nil {
+						return !apierrors.IsNotFound(err), err
+					}
+					var names []string
+					for _, ns := range sinkNsList {
+						names = append(names, ns.Name)
+						if ns.Name == expectedSinkNsName {
+							return true, nil
+						}
+					}
+					return false, fmt.Errorf("No namespace in sink cluster: %v", names)
+				}, 30*time.Second); err != nil {
+					t.Errorf("did not see namespace spec updated on sink cluster: %v", err)
+					return
+				}
+
+				if err := cowboyExpecter.ExpectBefore(ctx, func(ctx context.Context) (bool, error) {
```

Comment:
The change proposed in this PR involves the following:

1. transform `metadata.name` of synced namespaces
1. transform `metadata.namespace` of synced resources

Maybe the second change suggests checking that the cowboy resource has the expected transformed namespace?

Reply:
Yes this was the intention, but @ncdc pointed out that I wasn't setting the correct namespace when creating the lister used in this expecter.  So I'm going to update the lister to use the target namespace (on the pcluster).